### PR TITLE
Add "propRegex" option to only transform matching prop names

### PR DESCRIPTION
**Pull request by dounan**
This gives developers more control over which props are transformed. The main purpose is to use this to avoid transforming render callbacks.

### CODE REVIEW

File: babel/index.js [hunk 1]
```diff
@@ -79,6 +81,7 @@ module.exports = function(opts) {
         return;
       }
       logger.setLevel(log);
+      _propRegex = propRegex != null ? new RegExp(propRegex) : undefined;
```
Comment (zgotsch) [inline babel/index.js]: `_propRegexCompiled`?

File: babel/index.js [hunk 2]
```diff
@@ -151,6 +156,12 @@ module.exports = function(opts) {
     },
   };
 
+  function shouldSkipProp(name) {
+    // - Skip "ref" props.
+    // - If _propRegex is provided, skip props that don't match.
+    return name === "ref" || (_propRegex && !_propRegex.test(name));
```
Comment (zgotsch) [inline babel/index.js]: At the highest log level, should it log that it's skipping each prop that's skipped? Could be useful for debugging.

Comment (dounan) [reply babel/index.js]: I think that could be useful but it would require us to refactor the code so that we only log it if there is an actual transformation that can be done for the value inside the `JSXExpressionContainer` instead of any prop. Otherwise we'd be logging excessively. At this point, I'm not sure this feature is worth the refactor, but if this is something that people ask for in the future, we can add it then. What do you think?

File: babel/index.js [hunk 3]
```diff
@@ -158,8 +159,10 @@ module.exports = function(opts) {
 
   function shouldSkipProp(name) {
     // - Skip "ref" props.
-    // - If _propRegex is provided, skip props that don't match.
-    return name === "ref" || (_propRegex && !_propRegex.test(name));
+    // - If _propRegexCompiled is provided, skip props that don't match.
```
Comment (zgotsch) [inline babel/index.js]: `s/_propRegexCompiled/the "propRegex" option/`

Comment (dounan) [reply babel/index.js]: Good catch. I replaced "provided" with "exists" instead to avoid the assumption that only the "propRegex" option can create a `_propRegexCompiled` value

File: babel/index.js [hunk 4]
```diff
@@ -158,8 +159,10 @@ module.exports = function(opts) {
 
   function shouldSkipProp(name) {
     // - Skip "ref" props.
-    // - If _propRegex is provided, skip props that don't match.
-    return name === "ref" || (_propRegex && !_propRegex.test(name));
+    // - If _propRegexCompiled exists, skip props that don't match.
```
[the remark this answers is not in the document]
Comment (dounan) [reply babel/index.js]: Agreed. The reason I added it was because I thought the logic was a bit hard to parse and I figured a comment could help. I'm happy to take the comment out.
